astronomer/providers/google/cloud/hooks/dataproc.py

```diff
@@ -1,31 +1,111 @@
 import warnings
+from abc import ABC
 from typing import Any, Optional, Sequence, Tuple, Union
 
-from airflow.providers.google.cloud.hooks.dataproc import DataprocHook
+from airflow.providers.google.common.consts import CLIENT_INFO
 from airflow.providers.google.common.hooks.base_google import GoogleBaseHook
-from asgiref.sync import sync_to_async
+from google.api_core import gapic_v1
+from google.api_core.client_options import ClientOptions
 from google.api_core.retry import Retry
-from google.cloud.dataproc_v1 import Job
+from google.cloud.dataproc_v1 import (
+    ClusterControllerAsyncClient,
+    Job,
+    JobControllerAsyncClient,
+)
+from google.cloud.dataproc_v1.types import clusters
 
 JobType = Union[Job, Any]
 
 
-class DataprocHookAsync(DataprocHook):
-    """
-    Async Hook for Google Cloud Dataproc APIs.
-    All the methods in the hook where project_id is used must be called with
-    keyword arguments rather than positional.
-    """
+class DataprocHookAsync(GoogleBaseHook, ABC):
+    """Async Hook for Google Cloud Dataproc APIs"""
+
+    def get_cluster_client(
+        self, region: Optional[str] = None, location: Optional[str] = None
+    ) -> ClusterControllerAsyncClient:
+        """
+        Return ClusterControllerAsyncClient
```

Comment:
Remove the return attribute in the `docstring`, its not needed. This is present at multiple places.

Reply:
Modified